libs/payments/cart/src/lib/cart.service.ts

```diff
@@ -389,6 +393,45 @@ export class CartService {
     await this.cartManager.updateFreshCart(cartId, version, cartDetails);
   }
 
+  /**
+   * Fetch the invoice preview for the latest invoice associated with a cart
+   */
+  async previewLatestInvoice(
```

Comment:
That's a nice way of doing this, will update

Reply:
due to the way everything may-or-may-not-be `undefined`, trying to avoid a wildly nested `if/else` block, I've invented something equally bad! I present to you: **_ternary hell_**

```
const subscription = cart.stripeSubscriptionId
  ? await this.subscriptionManager.retrieve(cart.stripeSubscriptionId)
  : undefined;

const latestInvoice = subscription?.latest_invoice
  ? await this.invoiceManager.retrieve(subscription.latest_invoice)
  : undefined;

const paymentIntent = latestInvoice?.payment_intent
  ? await this.stripeClient.paymentIntentRetrieve(
      latestInvoice.payment_intent
    )
  : undefined;

const paymentMethodId =
  typeof paymentIntent?.payment_method === 'string'
    ? paymentIntent.payment_method
    : paymentIntent?.payment_method?.id;

const paymentMethod = paymentMethodId
  ? await this.paymentMethodManager.retrieve(paymentMethodId)
  : undefined;

const latestInvoicePreview = latestInvoice?.id
  ? await this.invoiceManager.previewInvoice(latestInvoice.id)
  : undefined;
const last4 = paymentMethod?.card?.last4;

if (latestInvoicePreview) {
  latestInvoicePreview.last4 = last4;
}
```

Not sure this is better, lets chat through options tomorrow